base_exception/tests/purchase_test.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2016 Akretion Mourad EL HADJ MIMOUNE
+# Copyright 2020 Hibou Corp.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class PurchaseTest(models.Model):
+    _inherit = "base.exception"
+    _name = "base.exception.test.purchase"
+    _description = "Base Ecxeption Test Model"
+
+    name = fields.Char(required=True)
+    user_id = fields.Many2one("res.users", string="Responsible")
+    state = fields.Selection(
+        [
+            ("draft", "New"),
+            ("cancel", "Cancelled"),
+            ("purchase", "Purchase"),
+            ("to approve", "To approve"),
+            ("done", "Done"),
+        ],
+        string="Status",
+        readonly=True,
+        default="draft",
+    )
+    active = fields.Boolean(default=True)
+    partner_id = fields.Many2one("res.partner", string="Partner")
+    line_ids = fields.One2many("base.exception.test.purchase.line", "lead_id")
+    amount_total = fields.Float(compute="_compute_amount_total", store=True)
+
+    @api.depends("line_ids")
+    def _compute_amount_total(self):
+        for record in self:
+            for line in record.line_ids:
+                record.amount_total += line.amount * line.qty
```

Comment:
Do we need to add a record.amount_total=False if the record have no line_ids?

Reply:
No, because the field (`amount_total`) is stored, which means you won't get a cache miss exception even if you comment out the `line_ids` in test_base_exception.TestBaseException.test_purchase_order_exception().  If you remove the `store=True` on the field AND comment out the `line_ids` in the test, then yes, you will get a cache miss and would need to provide other mechanisms to compute it. 

For fixing it, I'd change the whole thing to just `record.amount_total = sum(l.amount * l.qty for l in record.line_ids)` which will work for the empty case as well as only writing once in the non-empty case.  I won't commit that as a change in the migration ticket though as I wouldn't want anything to delay the migration merge. ;)